frontend/src/types/API.ts

```diff
@@ -0,0 +1,4 @@
+export interface APIResponse {
+  status: "ok" | "error"
```

Comment:
Am I correct, that with the current code, this will never be "error"? 

Two things I personally like/do:
- I usually have aditional "notAsked"/"loading" states, which I'm not sure if they make sense for us here. See https://package.elm-lang.org/packages/krisajenkins/remotedata/latest/RemoteData#RemoteData for some previous work in a different ecosystem
- I'm wondering if we can type the response differently between success/error/etc so that it only has the correct fields.

Reply:
It can be error, we're just not often having to check for it because we send fetch requests when appropriate (only when logged in for example).

I don't think we can type them differently because (it's perhaps a bit misleading) this type is being driven by the current form of backend **successful** responses so there's no way to distinguish them before checking for these attributes.

However, I have just noticed that some use an `error` attribute where others use `message`. I'll revisit this and perhaps homogenise those in the backend.

Reply:
This prompted me to re-look at the backend and realise I'd somehow got it into my head that the response with `status: "error"` could occur when the actual HTTP status was OK. In that case yeah this can be revisited properly with separate types :+1: